app/src/main/java/com/teamagam/gimelgimel/app/message/view/SendGeographicMessageDialog.java

```diff
@@ -0,0 +1,153 @@
+        SendGeographicMessageDialog fragment = new SendGeographicMessageDialog();
+
+        Bundle args = new Bundle();
+        args.putParcelable(ARG_POINT_GEOMETRY, pointGeometry);
+        fragment.setArguments(args);
+
+        return fragment;
+    }
+
+    /**
+     * Utility for creation of {@link SendGeographicMessageDialog} with a {@link PointGeometry}
+     * bundled argument (split into primitives) and containing {@link Fragment} (target-fragment)
+     * to be used if necessary
+     *
+     * @param pointGeometry  - the point to pass as argument (as primitives)
+     * @param targetFragment - the containing fragment
+     * @return bundled fragment with the given {@link PointGeometry}
+     */
+    public static SendGeographicMessageDialog newInstance(PointGeometry pointGeometry,
+                                                          Fragment targetFragment) {
+        SendGeographicMessageDialog f = newInstance(pointGeometry);
+        f.setTargetFragment(targetFragment, 0 /*optional*/);
+        return f;
+    }
+
+    @Override
+    public void onAttach(Activity activity) {
+        super.onAttach(activity);
+        DaggerMessagesComponent.builder()
+                .applicationComponent(((GGApplication) getActivity().getApplication()).getApplicationComponent())
+                .messageModule(new MessageModule())
+                .build()
+                .inject(this);}
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+    }
+
+    @Override
+    public void onPause() {
+        super.onPause();
+    }
+
+    @Override
+    protected int getTitleResId() {
+        return R.string.dialog_send_message_geo_title;
+    }
+
+    @Override
+    protected int getDialogLayout() {
+        return R.layout.dialog_send_geo_message;
+    }
+
+    @Override
+    protected String getNegativeString() {
+        return getString(R.string.dialog_send_message_cancel);
+    }
+
+    @Override
+    protected String getPositiveString() {
+        return getString(R.string.dialog_send_message_ok);
+    }
+
+    @Override
+    protected View onCreateDialogLayout() {
+        DialogSendGeoMessageBinding binding = DataBindingUtil.inflate(
+                LayoutInflater.from(getActivity()), getDialogLayout(),
+                null, false);
+        PointGeometry point = getArguments().getParcelable(ARG_POINT_GEOMETRY);
+        mViewModel.init(this, point);
+        binding.setViewModel(mViewModel);
+
+        return binding.getRoot();
+    }
+
+    @Override
+    protected boolean hasNegativeButton() {
+        return true;
+    }
+
+    @Override
+    protected boolean hasPositiveButton() {
+        return true;
+    }
+
+    @Override
+    protected void onPositiveClick() {
+        sLogger.userInteraction("Clicked OK");
+        mViewModel.clickedOK();
```

Comment:
I like to call the method after what they do.
We'll change it because of our discussion, but "clickedOK()" isn't a good name.
Rename it to what the method does, because we know the user clicked ok (The overriden method name).